cmd/client-fs.go

```diff
@@ -409,9 +409,13 @@ func (f *fsClient) put(reader io.Reader, size int64, metadata map[string][]strin
 
 // Put - create a new file with metadata.
 func (f *fsClient) Put(ctx context.Context, reader io.Reader, size int64, metadata map[string]string, progress io.Reader, sse encrypt.ServerSide, md5, disableMultipart bool) (int64, *probe.Error) {
-	if metadata["mc-attrs"] != "" {
+	metaDataKey := "mc-attrs"
+	if metadata["X-Amz-Meta-Mc-Attrs"] != "" {
+		metaDataKey = "X-Amz-Meta-Mc-Attrs"
+	}
+	if metadata[metaDataKey] != "" {
 		meta := make(map[string][]string)
-		meta["mc-attrs"] = append(meta["mc-attrs"], metadata["mc-attrs"])
+		meta["mc-attrs"] = append(meta["mc-attrs"], metadata[metaDataKey])
```

Comment:
```suggestion
		meta[metaDataKey] = append(meta[metaDataKey], metadata[metaDataKey])
```

Reply:
@harshavardhana,
I did it this way, because I did not want to cause any regression, in case the same assumption is made in some other caller functions. If so, I also need to change those part of the codes in the same way.
Do you still want to see your suggestion to be implemented?

Reply:
it looks a bit confusing @ebozduman we need to use a single metadata key.

Reply:
@harshavardhana, you suggested 2 ways:

1. I've just tested with your suggestion in your first comment.
As I expected, there is at least one or more callers which also expect the metadata key to be `"mc-attrs"`, hence the fix doesn't work. To make it work, there has to be more changes in this PR, if we choose to go this way.

2. Single key sounds better to me also
I can replace `"mc-attrs"` key with `"X-Amz-Meta-Mc-Attrs"` every where. (I'd still use a variable).
But, the only thing I am not %100 sure is that if there is any scenario where the key comes as `"mc-attrs"` and if we miss that code path, this PR is going to cause regression.

3. My original change (confusing, short and fastest solution)

Please let me know which option you want to go.

Reply:
After discussing with @kannappanr and @poornas, we've decided to go with single key, `"X-Amz-Meta-Mc-Attrs"` everywhere. This way, we also become S3 compliant.